monitoring/bin/deploy_monitoring_tenant_openshift.sh

```diff
@@ -0,0 +1,258 @@
+
+log_info "Deploying Grafana..."
+userGrafanaYAML=$TMP_DIR/empty.yaml
+if [ -f "$USER_DIR/monitoring/user-values-openshift-grafana-$VIYA_TENANT.yaml" ]; then
+  userGrafanaYAML="$USER_DIR/monitoring/user-values-openshift-grafana-$VIYA_TENANT.yaml"
+  log_debug "User response file for Grafana found at [$userGrafanaYAML]"
+fi
+
+if ! helm3ReleaseExists v4m-grafana-$VIYA_TENANT $VIYA_NS; then
+  firstTimeGrafana=true
+fi
+
+OPENSHIFT_AUTH_ENABLE=${OPENSHIFT_AUTH_ENABLE:-true}
+log_debug "Creating the Prometheus service to generate TLS certs..."
+kubectl apply -n $VIYA_NS -f $tenantDir/openshift/v4m-prometheus-tenant-svc.yaml
+log_debug "Creating the Grafana service to generate TLS certs..."
+kubectl apply -n $VIYA_NS -f $tenantDir/openshift/v4m-grafana-tenant-svc.yaml
+grafanaAuthYAML="monitoring/openshift/grafana-proxy-values.yaml"
+
+if [ -n "$GRAFANA_ADMIN_PASSWORD" ]; then
+  grafanaPwd="--set adminPassword=$GRAFANA_ADMIN_PASSWORD"
+fi
+
+# Optional workload node placement support
+MON_NODE_PLACEMENT_ENABLE=${MON_NODE_PLACEMENT_ENABLE:-${NODE_PLACEMENT_ENABLE:-false}}
+if [ "$MON_NODE_PLACEMENT_ENABLE" == "true" ]; then
+  log_info "Enabling monitoring components for workload node placement"
+  wnpValuesFile="monitoring/node-placement/values-grafana-wnp.yaml"
+else
+  log_debug "Workload node placement support is disabled"
+  wnpValuesFile="$TMP_DIR/empty.yaml"
+fi
+
+if [ "$OPENSHIFT_PATH_ROUTES" == "true" ]; then
+  OPENSHIFT_ROUTE_PATH_GRAFANA=${OPENSHIFT_ROUTE_PATH_GRAFANA:-/grafana}
+fi
+if [ "$OPENSHIFT_ROUTE_PATH_GRAFANA" != "" ] && [ "$OPENSHIFT_ROUTE_PATH_GRAFANA" != "/" ]; then
+  grafanaSubPath="true"
+else
+  grafanaSubPath="false"
+fi
+
+log_info "Deploying Grafana..."
+grafanaYAML=$tenantDir/openshift/mt-grafana-openshift-values.yaml
+OPENSHIFT_GRAFANA_CHART_VERSION=${OPENSHIFT_GRAFANA_CHART_VERSION:-6.17.2}
+helm upgrade --install $helmDebug \
+  -n "$VIYA_NS" \
+  -f "$wnpValuesFile" \
+  -f "$grafanaYAML" \
+  -f "$grafanaAuthYAML" \
+  -f "$userGrafanaYAML" \
+  --set 'grafana\.ini'.server.domain=$OPENSHIFT_ROUTE_DOMAIN \
+  --set 'grafana\.ini'.server.root_url=https://v4m-grafana-$VIYA_NS.$OPENSHIFT_ROUTE_DOMAIN$OPENSHIFT_ROUTE_PATH_GRAFANA \
+  --set 'grafana\.ini'.server.serve_from_sub_path=$grafanaSubPath \
+  --version "$OPENSHIFT_GRAFANA_CHART_VERSION" \
+  --atomic \
+  $grafanaPwd \
+  $extraArgs \
+  v4m-grafana-$VIYA_NS-$VIYA_TENANT \
+  grafana/grafana
+
+if [ "$OPENSHIFT_AUTH_ENABLE" == "true" ]; then
+  log_info "Using OpenShift authentication for [$VIYA_TENANT] tenant instance of Grafana"
+  log_debug "Annotating grafana-serviceaccount-$VIYA_TENANT"
+  kubectl annotate serviceaccount -n $VIYA_NS --overwrite grafana-serviceaccount-$VIYA_TENANT 'serviceaccounts.openshift.io/oauth-redirectreference.primary={"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"v4m-grafana-'$VIYA_TENANT'"}}'
```

Comment:
Will the OpenShift authentication ensure that tenant admins can ONLY access the tenant-specific instance of these monitoring components and nothing else?  How are tenant admins limited within the OpenShift auth configuration?